### DISCUSSION TITLE
Help with POST

### DISCUSSION THREAD

**Discussion by magick93** [Orange-OpenSource/hurl#3061] [Q&A] [answered]
Hello

I'm trying to run the following:

```
POST https://my.service.domain/
[BasicAuth]
token: hello_world
```json
{
  "category_details": {
    "category_name": "Some Category",
    "parent_category_name": null
  },
  "business_model_name": "Some Business Model",
  "industry_name": "Some Industry",
  "business_structure_name": "Some Business Structure"
}
```

```

If I dont include the basic auth, I get `Unorthorized` which is expected.
However, If I _do_ include it, I get `Bad Request`. 

I'm obviously not constructing the request properly but I cannot understand what I'm doing wrong. 

# Version info:
```
hurl 4.3.0 (x86_64-pc-linux-gnu) libcurl/8.8.0 OpenSSL/3.3.1 zlib/1.3.1 brotli/1.1.0 zstd/1.5.6 libidn2/2.3.7 libssh2/1.11.0 nghttp2/1.61.0
Features (libcurl):  alt-svc AsynchDNS brotli HSTS HTTP2 IDN IPv6 Largefile libz NTLM SPNEGO SSL TLS-SRP UnixSockets zstd
Features (built-in): brotli
```

**Comment by jcamiel** [comment, marked as answer]
Hi @magick93 

The 400 error code `Bad request` indicates that there could be an error with the requests headers, or body. 

First of all, you can run your script with `--verbose`. This way, you can check that the requests headers are those expected. You can see also the `curl` command line in Hurl debug logs. You can try this `curl` command line to check that you've the same error (in case of a potential bug in Hurl).

You have also `--very-verbose` option that give you more details (mainly libcurl logs).

Another possibility, instead of this syntax:

~~~hurl
POST https://my.service.domain/
[BasicAuth]
token: hello_world
```json
{
  "category_details": {
    "category_name": "Some Category",
    "parent_category_name": null
  },
  "business_model_name": "Some Business Model",
  "industry_name": "Some Industry",
  "business_structure_name": "Some Business Structure"
}
```
~~~

You can try this syntax (JSON body instead of multiline string body):

```hurl
POST https://my.service.domain/
[BasicAuth]
token: hello_world
{
  "category_details": {
    "category_name": "Some Category",
    "parent_category_name": null
  },
  "business_model_name": "Some Business Model",
  "industry_name": "Some Industry",
  "business_structure_name": "Some Business Structure"
}
```

The second syntax is equivalent but we're more strict in the request body JSON parsing.
 
If you still have the problem, instead of an `[BasicAuth]` section, you can try a `--user` option

```hurl
POST https://my.service.domain/
[Options]
user: login:password
{
  "category_details": {
    "category_name": "Some Category",
    "parent_category_name": null
  },
  "business_model_name": "Some Business Model",
  "industry_name": "Some Industry",
  "business_structure_name": "Some Business Structure"
}
```

These are equivalent way of doing basic auth but there could still be a bug in Hurl...

Hope it's helping you!
